packages/front-end/components/Features/DraftModal.tsx

```diff
@@ -97,11 +110,15 @@ export default function DraftModal({
             title: `Rules - ${env.id}`,
             a: JSON.stringify(
               feature.environmentSettings?.[env.id]?.rules || [],
-              null,
+              nestedJSONReplacer,
```

Comment:
I don't know if using a replacer here is the right approach.  Replacers are applied recursively, so if the feature value is a JSON string that itself has a key called `value`, it will try to recursively expand it.  There are only 2 places where values are specified in a rule, so explicitly JSON parsing those might be safer.

Something like this:
```ts
import cloneDeep from "lodash/cloneDeep";

function jsonParseSafe(value: string): unknown {
  try {
    return JSON.parse(value);
  } catch (e) {
    // If parse error, return the original value
    return value;
  }
}

function expandValues(rules: FeatureRule[]) {
  return cloneDeep(rules).map((rule) => {
    // Multiple values (A/B test rules)
    if ("values" in rule) {
      return {
        ...rule,
        values: rule.values.map((variation) => ({
          ...variation,
          value: jsonParseSafe(variation.value),
        })),
      };
    }
    // Single value (Force and Rollout rules)
    else {
      return {
        ...rule,
        value: jsonParseSafe(rule.value),
      };
    }
  });
}

// When doing the diff
a: JSON.stringify(
  expandValues(feature.environmentSettings?.[env.id]?.rules || []),
  null,
  2
)
```